### PR TITLE
[CAS-538] Add support for pinned messages to LLC

### PR DESCRIPTION
**Pull request by adasiewiczr**
https://stream-io.atlassian.net/browse/CAS-538

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Message.kt [hunk 1]
```diff
@@ -94,6 +94,14 @@ public data class Message(
...
-    var replyMessageId: String? = null
+    var replyMessageId: String? = null,
+
+    var pinned: Boolean = false,
+    @SerializedName("pinned_at")
+    var pinnedAt: Date? = null,
+    @SerializedName("pin_expires")
+    var pinExpires: Date? = null,
+    @SerializedName("pinned_by")
+    var pinnedBy: User? = null,
```
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Message.kt:105]: I assume we should also update `MessageEntity` and `ChannelEntity`?

**Review by zsmb13 — COMMENTED** [review]
The JS implementation makes a bunch of updates to their channel state as message events (like `message.new`, `message.updated, `message.deleted` are coming in. Don't we have similar updates to perform?

On another note, is there anything here that we can add tests for?

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Message.kt (hunk 1 above)
Comment (adasiewiczr) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Message.kt:105]: As far as I know, for now, we are going to only support this in LLC so I'm not sure if there is a need to update those entities

Comment (zsmb13) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Message.kt:105]: Ah, okay, I didn't realize the scope of the task. The PR's title is actually clear, the Jira task perhaps not so much.

**Comment by adasiewiczr** [comment]
@zsmb13 I'm not sure about the tests because basically, I've added a bunch of new fields and API calls which should return updated objects

**Review by zsmb13 — APPROVED** [review]
As far as covering the LLC, this LGTM 👍

**Pull request merged**